src/shortcut.rs

```diff
@@ -0,0 +1,52 @@
+#[allow(unused_imports)]
+use std::ffi::CString;
+use std::path::PathBuf;
+#[allow(unused_imports)]
+use std::process::Command;
+
+#[cfg(target_os = "windows")]
+#[link(name = "shell32")]
+extern "system" {
+    fn ShellExecuteA(
+        hwnd: isize,
+        lpOperation: *const i8,
+        lpFile: *const i8,
+        lpParameters: *const i8,
+        lpDirectory: *const i8,
+        nShowCmd: i32,
+    ) -> isize;
+}
+
+pub fn open_folder(path: PathBuf) {
+    #[cfg(target_os = "linux")]
+    {
+        Command::new("xdg-open")
+            .arg(path.to_str().unwrap())
+            .spawn()
+            .expect("Failed to Open Folder");
+    }
+    #[cfg(target_os = "windows")]
+    {
+        let operation = "open\0";
```

Comment:
In Rust you should create `CString` or `CStr` instead of padding with zero bytes manually:
```suggestion
        let operation = c"open";
```

See https://doc.rust-lang.org/nightly/edition-guide/rust-2021/c-string-literals.html